### ISSUE TITLE
another fatal, uncaught timeout-related exception in mscolab upon activating an operation

### ISSUE THREAD

**Issue by joernu76** [Open-MSS/MSS#1969]
```
2023-08-30 08:49:24,850 (qt.excepthook:183): CRITICAL: MSS Version: 8.2.0
2023-08-30 08:49:24,850 (qt.excepthook:184): CRITICAL: Python Version: 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]
2023-08-30 08:49:24,857 (qt.excepthook:185): CRITICAL: Platform: Linux-6.1.0-10-amd64-x86_64-with-glibc2.36 (('64bit', 'ELF'))
2023-08-30 08:49:24,857 (qt.excepthook:186): CRITICAL: Fatal error: Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 467, in _make_request
    self._validate_conn(conn)
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1092, in _validate_conn
    conn.connect()
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connection.py", line 635, in connect
    sock_and_verified = _ssl_wrap_socket_and_match_hostname(
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connection.py", line 776, in _ssl_wrap_socket_and_match_hostname
    ssl_sock = ssl_wrap_socket(
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/util/ssl_.py", line 466, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/util/ssl_.py", line 510, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/opt/anaconda3/lib/python3.10/ssl.py", line 513, in wrap_socket
    return self.sslsocket_class._create(
  File "/opt/anaconda3/lib/python3.10/ssl.py", line 1071, in _create
    self.do_handshake()
  File "/opt/anaconda3/lib/python3.10/ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
TimeoutError: _ssl.c:990: The handshake operation timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 844, in urlopen
    retries = retries.increment(
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/util/retry.py", line 470, in increment
    raise reraise(type(error), error, _stacktrace)
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/util/util.py", line 39, in reraise
    raise value
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 790, in urlopen
    response = self._make_request(
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 491, in _make_request
    raise new_e
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 469, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)
  File "/opt/anaconda3/lib/python3.10/site-packages/urllib3/connectionpool.py", line 370, in _raise_timeout
    raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='forecast.fz-juelich.de', port=443): Read timed out. (read timeout=2)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/icg173/src/mss/mslib/msui/mscolab.py", line 1650, in set_active_op_id
    requests.post(f'{self.mscolab_server_url}/set_last_used', data=data,
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/sessions.py", line 702, in send
    r = adapter.send(request, **kwargs)
  File "/opt/anaconda3/lib/python3.10/site-packages/requests/adapters.py", line 532, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='forecast.fz-juelich.de', port=443): Read timed out. (read timeout=2)

```

**Comment by joernu76** [comment]
In develop.

**Comment by ReimarBauer** [comment]
which requests version is that? This  "requests.post(f'{self.mscolab_server_url}/set_last_used'" should be moved to a data base operation on the server site. Currently this is executed on login and it is completly on the wrong place.

While the client was calling this function there was a problem with the network

**Comment by joernu76** [comment]
I have no idea about the logic/intention of the code. I prepared a fix adding more try/excepts to prevent the Exception and forcing a log-out.

**Comment by joernu76** [comment]
so the merge from stable to develop failed?

**Comment by joernu76** [comment]
I just made a diff and I see several places that seems to have been mis-merged:
* after-login identified by you
* the try/except in handle_leave_operation is missing as well.
* the try/except in rename_operation is missing 
Do you see more things? What I can judge, it seems fine.

**Comment by joernu76** [comment]
The after-login stuff was fixed by me in stable by 78d5f1ae3b7efd00d33984f7bf0850663eaba0f4 on 14 Aug 2023.

After trying to replicate the merge from stable to develop from Aug 14, I would say that it was botched due to the serious conflicts. At least the problem with the after-login can be traced there.
